testnet/stacks-node/src/neon_node.rs

```diff
@@ -568,8 +572,15 @@ fn spawn_miner_relayer(
                         &mut mem_pool,
                         burn_fee_cap,
                         &mut bitcoin_controller,
+                        &last_mined_blocks,
```

Comment:
For this `last_mined_blocks` tracking to work, `RunTenure` events would need to issued every time the Stacks chain tip updates (or, at least multiple times per sortition). The way that the neon_node and neon run loop were previously structured, the `relayer_issue_tenure` function, which sends that event, is only invoked once per sortition. I can't tell how that's changed in `runloop/neon.rs` file -- it still seems like the `burnchain.sync()` function is invoked (which waits until a sortition has processed) before invoking `relayer_issue_tenure` a second time.

Reply:
Ah -- this is where `poll_time_secs` comes into play per your earlier comment.  If you look at `run_loop/neon.rs`, you should see that the loop can now multiple passes per Bitcoin epoch.  In particular, the `relayer_issue_tenure()` method gets invoked each time the PoX watchdog does a burnchain poll.  If the sortition DB's height matches the burnchain's height, the node proceeds to mine (however, this will need to be patched before merging -- per @psq's insight earlier today, the node should only attempt to mine if it's not in the initial block download phase lest it begin mining too early).

Reply:
`burnchain.sync()` blocks waiting for the next sortition (or it at least appears to), which would prevent the loop from iterating.

Reply:
But the loop _does_ iterate.

Reply:
I can't conclude that from looking at the code, though (and there's no tests of this behavior). Do you see the RBF code getting exercised when you run this locally? If so, can you provide the commands you ran to start up the network, and how you verified that RBFs are getting issued?

Reply:
You can definitely see that `relayer_issue_tenure` gets invoked multiple times per Bitcoin block -- I had to patch the `RelayerDirective::RunTenure` case so that `bump_processed_counter()` only gets called once, for example.  Here's the relevant log output from running one of these tests:

```
INFO [1603053105.219] [testnet/stacks-node/src/neon_node.rs:1016] [ThreadId(9)] Stacks block assembled: bc3ae07b8d4ed2dc9938957b98e4ef32595455c31596563f0b3b88638c855d87, with 2 txs, attempt 1
DEBUG [1603053105.223] [testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs:1093] [ThreadId(9)] BitcoinRPC builder: Some("neon-tester"):Some("neon-tester-pass")@http://127.0.0.1:8332/
INFO [1603053105.243] [testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs:611] [ThreadId(9)] Miner node: submitting leader_block_commit op - 04ee0b1602eb18fef7986887a7e8769a30c9df981d33c8380d255edef003abdcd243a0eb74afdf6740e6c423e62aec631519a24cf5b1d62bf8a3e06ddc695dcb77
DEBUG [1603053105.243] [testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs:1093] [ThreadId(9)] BitcoinRPC builder: Some("neon-tester"):Some("neon-tester-pass")@http://127.0.0.1:8332/
DEBUG [1603053105.246] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603053105.248] [testnet/stacks-node/src/neon_node.rs:906] [ThreadId(9)] Mining tenure's last consensus hash: 2c0cbb80bf7ff4ea2e5a92d9f750878b3ae1cb09 (height 207 hash 635b620d1dd9d1e3975531da457563b3c053de2e7cb556c2c4a515a17e1921c2), stacks tip consensus hash: 2c0cbb80bf7ff4ea2e5a92d9f750878b3ae1cb09 (height 207 hash 635b620d1dd9d1e3975531da457563b3c053de2e7cb556c2c4a515a17e1921c2)
DEBUG [1603053105.250] [testnet/stacks-node/src/neon_node.rs:956] [ThreadId(9)] Stacks tip is unchanged since we last tried to mine a block (2c0cbb80bf7ff4ea2e5a92d9f750878b3ae1cb09/bc3ae07b8d4ed2dc9938957b98e4ef32595455c31596563f0b3b88638c855d87 at height 5 with 2 txs, in 635b620d1dd9d1e3975531da457563b3c053de2e7cb556c2c4a515a17e1921c2 at burn height 207)
DEBUG [1603053105.250] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603053105.253] [testnet/stacks-node/src/neon_node.rs:906] [ThreadId(9)] Mining tenure's last consensus hash: 2c0cbb80bf7ff4ea2e5a92d9f750878b3ae1cb09 (height 207 hash 635b620d1dd9d1e3975531da457563b3c053de2e7cb556c2c4a515a17e1921c2), stacks tip consensus hash: 2c0cbb80bf7ff4ea2e5a92d9f750878b3ae1cb09 (height 207 hash 635b620d1dd9d1e3975531da457563b3c053de2e7cb556c2c4a515a17e1921c2)
DEBUG [1603053105.254] [testnet/stacks-node/src/neon_node.rs:956] [ThreadId(9)] Stacks tip is unchanged since we last tried to mine a block (2c0cbb80bf7ff4ea2e5a92d9f750878b3ae1cb09/bc3ae07b8d4ed2dc9938957b98e4ef32595455c31596563f0b3b88638c855d87 at height 5 with 2 txs, in 635b620d1dd9d1e3975531da457563b3c053de2e7cb556c2c4a515a17e1921c2 at burn height 207)
DEBUG [1603053105.254] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603053105.257] [testnet/stacks-node/src/neon_node.rs:906] [ThreadId(9)] Mining tenure's last consensus hash: 2c0cbb80bf7ff4ea2e5a92d9f750878b3ae1cb09 (height 207 hash 635b620d1dd9d1e3975531da457563b3c053de2e7cb556c2c4a515a17e1921c2), stacks tip consensus hash: 2c0cbb80bf7ff4ea2e5a92d9f750878b3ae1cb09 (height 207 hash 635b620d1dd9d1e3975531da457563b3c053de2e7cb556c2c4a515a17e1921c2)
DEBUG [1603053105.258] [testnet/stacks-node/src/neon_node.rs:956] [ThreadId(9)] Stacks tip is unchanged since we last tried to mine a block (2c0cbb80bf7ff4ea2e5a92d9f750878b3ae1cb09/bc3ae07b8d4ed2dc9938957b98e4ef32595455c31596563f0b3b88638c855d87 at height 5 with 2 txs, in 635b620d1dd9d1e3975531da457563b3c053de2e7cb556c2c4a515a17e1921c2 at burn height 207)
...
```

You can tell that RBF is working because you don't see any rejected block-commits.  If RBF was not working, then you'd see multiple conflicting block-commits get processed and rejected (i.e. src/burnchains/burnchain.rs#L680 would get executed).  As for actually triggering the RBF code-path that re-sends the transaction, you'll need to run two miners concurrently to see it, and get them to race each other.  A miner will download a Bitcoin block and process its sortitions, detect that it lost the sortition, attempt to mine its own block, receive the winning block, detect that the Stacks chain tip has changed, re-mine a Stacks block, and replace its block-commit transaction with RBF.  You'd see this in the log file with something like:

```
INFO [1603053105.219] [testnet/stacks-node/src/neon_node.rs:1016] [ThreadId(9)] Stacks block assembled: bc3ae07b8d4ed2dc9938957b98e4ef32595455c31596563f0b3b88638c855d87, with 2 txs, attempt 1
...
INFO [1603053115.219] [testnet/stacks-node/src/neon_node.rs:1016] [ThreadId(9)] Stacks block assembled: 9e47e70f5e50f03ac2da6654a6a303e7b78a87fd93c9755d9bcf8a30d1cc691f, with 2 txs, attempt 2
...
INFO [1603053125.219] [testnet/stacks-node/src/neon_node.rs:1016] [ThreadId(9)] Stacks block assembled: 635b620d1dd9d1e3975531da457563b3c053de2e7cb556c2c4a515a17e1921c2, with 2 txs, attempt 3
...
```

Reply:
If you want to see it for yourself, run `cd net-test/bin && ./start.sh master` in one terminal, and then `cd net-test/bin && ./start.sh miner` in another, and watch `  You'll want to put `faucet.sh` in your `$PATH` first.  You can watch the outputs with `tail -f net-test/mnt/stacks-node-master.log` and `tail -f net-test/mnt/stacks-node-miner.log`.

Reply:
Isn't that log output somewhat concerning? It looks like it's being invoked multiple times in the span of a second.

Reply:
I think this can happen if the relayer's communication channel is processing tenures more slowly than sortitions are being processed -- multiple `RunTenure` events will be stored in the channel.

Reply:
Yes -- this happens in the `size_check_integration_test`, where processing a block is indeed time-consuming.

Reply:
But that isn't the behavior we're trying to achieve, right? That only occurs when the relayer thread is processing tenures more slowly than sortitions are happening. It seems like the intent of the `last_mined_blocks` and RBF behavior is that it be invoked when the miner produces a block commit more quickly than an incoming Stacks block is processed (so it can update the chain tip it mines off of). I think this PR fixes some of the forking problems that the testnet is seeing by improving the block relay, but I don't think this RBF behavior (as it currently stands) could make an impact.

Reply:
The behavior we're trying to achieve is that if two miners A and B are online, and (w.l.o.g) A wins sortition in round X and B wins sortition in round X+1, then B builds on top of A's block if A broadcasts it during round X.  The problem this PR solves is removing this race condition:

1. Miner A wins sortition
2. Miner B creates a Stacks block and sends a block-commit
3. Miner B receives miner A's block. Without this PR, this introduces a short fork.
4. (with this PR) miner B re-creates a Stacks block and RBF's its block-commit

The relay fixes alone do make it less likely that this race condition occurs (i.e. event 3 happens less), but without the additional RBF fix, these short-lived forks can still occur.  If you assume that block arrival times are random during the sortition's epoch, then without RBF, if miner B waits for fraction F of sortition X's time before building a block, then miner B will produce a fork with probability 1 - F.  With RBF, B produces a fork with probability _e_, where _e_ is the fraction of the sortiton's epoch time taken for the Stacks block to propagate from A to B and for B to produce a new Stacks block and propagate a RBF block-commit.  RBF is an improvement if _e < 1 - F_, and unlike before, we don't have to guess what _e_ is in advance (whereas without RBF, we have to guess what a good value of _F_ is).

If the full network test suite was ready, I'd implement an integration test that had 2 miners work together on mining a chain for 100 blocks, I'd set the `wait_time_for_microblocks` to 0, and and fail if the canonical chain was less than 95 blocks long.  If you try out the code, you'll see that 2 miners can reliably build a 100-block chain after 100 sortitions _even with `wait_time_for_microblocks` set to 0._

Reply:
Yes, that's the scenario I think this is supposed to address, but am not convinced that it does (or if it does, does so accidentally). In the current code, there is one RunTenure event created per sortition processed. If a RunTenure event is processed multiple times during a single sortition, it would be because the miner fell behind. I will look at getting these scripts to run on my local machine (it will take some time, because they don't run on my device natively, so I'll have to make docker containers for these), but the logs you showed do not demonstrate the race condition described (rather, they show a miner who has fallen behind in processing tenures).

Reply:
>  If a RunTenure event is processed multiple times during a single sortition, it would be because the miner fell behind.

This is no longer true in this PR -- this PR intentionally changed this behavior so that a single sortition may generate multiple `RunTenure`s (i.e. at a rate of once per `poll_time_secs`, as long as `poll_time_secs` isn't unreasonably small).  As expected, I am seeing `RunTenure` getting processed multiple times during a Bitcoin epoch, without any channel back-pressure, and without the miner falling behind.  In fact, examining the test logs further, I can see that the quick succession of `RelayerDirective::RunTenure` directives are all for the same block height -- no falling behind here.  You can see it with:

```
$ egrep 'Proceeding to mine block|Run tenure' /tmp/out.txt 
DEBUG [1603139323.487] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 202. Proceeding to mine blocks
DEBUG [1603139330.685] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 203. Proceeding to mine blocks
DEBUG [1603139331.185] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139338.972] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603139340.402] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139345.262] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603139345.762] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139352.467] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603139352.968] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139359.764] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603139360.264] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139367.432] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603139367.933] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139374.782] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603139375.282] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139382.745] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603139388.891] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603139394.880] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139396.683] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603139403.827] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603139410.985] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603139418.154] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603139421.642] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139421.647] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139421.651] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139421.655] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139421.659] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139426.764] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603139433.949] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603139440.237] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603139441.881] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139447.424] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603139454.578] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603139461.848] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603139466.129] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139466.133] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139466.137] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139466.141] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139466.146] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139469.855] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603139477.293] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603139478.990] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139488.974] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603139500.166] [testnet/stacks-node/src/run_loop/neon.rs:277] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603139503.883] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139503.889] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139503.893] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
```

(This is in `size_check_integration_test`'s output).

Now, consider the first long burst of `RelayerDirective::RunTenure`s, starting with the one at `1603139394.880`.  Looking at what `ThreadId(9)` (the relayer thread) was up to in-between `1603139394.880` and `1603139421.642`, I see the following:

```
DEBUG [1603139394.889] [testnet/stacks-node/src/neon_node.rs:906] [ThreadId(9)] Mining tenure's last consensus hash: fcff47e4ad8f793a8fc0cb9f062968acf754acbe (height 205 hash 4f4ba280b1e24625f8364722a172d7359bc122db77edaa0bb6e63f582101ae0d), stacks tip consensus hash: fcff47e4ad8f793a8fc0cb9f062968acf754acbe (height 205 hash 4f4ba280b1e24625f8364722a172d7359bc122db77edaa0bb6e63f582101ae0d)
DEBUG [1603139394.901] [testnet/stacks-node/src/neon_node.rs:983] [ThreadId(9)] Generated VRF Proof: 736255e231f3c2fcb845bc1ad748ceb33d6792fadc81d9e1baadd7a4f60db26a23648875ea98b14ee0bb04740d34c4bee7460876bcc76cd6034f5f841df2fe3da9dc2d685ffc9f9ad142296737706706 over 3973934ad5b2f4c24aacfc5cca9cbf3ae16b62dbe1b5d57451ec7f9a353ef791 with key 862842da1c0f18ab62ee54560080044a245c6acc96691488e3594ff3a9017937
DEBUG [1603139394.902] [src/chainstate/stacks/miner.rs:960] [ThreadId(9)] Build anchored block off of fcff47e4ad8f793a8fc0cb9f062968acf754acbe/1b82e31de5a46df0e464c2389624e42f77e9c52d5ca19dedac9616341bd128b2 height 2
DEBUG [1603139394.902] [src/chainstate/stacks/index/storage.rs:770] [ThreadId(9)] Opened TrieFileStorage /tmp/stacks-testnet-120b0b10e001ddb4/chainstate/chain-00000080-testnet/vm/index;
DEBUG [1603139394.903] [src/chainstate/stacks/db/blocks.rs:497] [ThreadId(9)] Opened blocks DB /tmp/stacks-testnet-120b0b10e001ddb4/chainstate/chain-00000080-testnet/blocks/staging.db
DEBUG [1603139394.904] [src/chainstate/stacks/index/storage.rs:770] [ThreadId(9)] Opened TrieFileStorage /tmp/stacks-testnet-120b0b10e001ddb4/chainstate/chain-00000080-testnet/vm/clarity/marf;
DEBUG [1603139394.904] [src/chainstate/stacks/index/storage.rs:770] [ThreadId(9)] Opened TrieFileStorage /tmp/stacks-testnet-120b0b10e001ddb4/chainstate/chain-00000080-testnet/vm/index;
DEBUG [1603139394.904] [src/chainstate/stacks/db/blocks.rs:497] [ThreadId(9)] Opened blocks DB /tmp/stacks-testnet-120b0b10e001ddb4/chainstate/chain-00000080-testnet/blocks/staging.db
DEBUG [1603139394.905] [src/chainstate/stacks/index/storage.rs:770] [ThreadId(9)] Opened TrieFileStorage /tmp/stacks-testnet-120b0b10e001ddb4/chainstate/chain-00000080-testnet/vm/clarity/marf;
DEBUG [1603139396.450] [src/chainstate/stacks/index/marf.rs:214] [ThreadId(9)] Extending off of existing node e56fde2bbb789fab4ad51501fc7658fbcb440b1075815d86022c17e78f21c883
DEBUG [1603139396.450] [src/chainstate/stacks/index/marf.rs:579] [ThreadId(9)] Extend e56fde2bbb789fab4ad51501fc7658fbcb440b1075815d86022c17e78f21c883 to b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b
DEBUG [1603139396.451] [src/chainstate/stacks/index/marf.rs:279] [ThreadId(9)] Set __MARF_BLOCK_HEIGHT_TO_HASH::3 = b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b
DEBUG [1603139396.451] [src/chainstate/stacks/index/marf.rs:283] [ThreadId(9)] Set __MARF_BLOCK_HASH_TO_HEIGHT::b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b = 3
DEBUG [1603139396.451] [src/chainstate/stacks/index/marf.rs:287] [ThreadId(9)] Set __MARF_BLOCK_HEIGHT_SELF = 3
DEBUG [1603139396.451] [src/chainstate/stacks/index/marf.rs:302] [ThreadId(9)] Set __MARF_BLOCK_HEIGHT_TO_HASH::2 = e56fde2bbb789fab4ad51501fc7658fbcb440b1075815d86022c17e78f21c883
DEBUG [1603139396.451] [src/chainstate/stacks/index/marf.rs:308] [ThreadId(9)] Set __MARF_BLOCK_HASH_TO_HEIGHT::e56fde2bbb789fab4ad51501fc7658fbcb440b1075815d86022c17e78f21c883 = 2
DEBUG [1603139396.452] [src/chainstate/stacks/index/marf.rs:853] [ThreadId(9)] MARF Insert in b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b: 'd3b3d3aa9f9a2fdd3636ae2b6600f414fbbaa0d1385568c99707d59f4526eff1' = '03000000000000000000000000000000000000000000000000000000000000000000000000000000' (...[])
DEBUG [1603139396.453] [src/chainstate/stacks/index/trie.rs:851] [ThreadId(9)] Next root hash is 0b717373b7f5cfece4c53ae29302eb3edaa2f93986286873bc97561fd897422b (update_skiplist=false)
DEBUG [1603139396.453] [src/chainstate/stacks/index/marf.rs:853] [ThreadId(9)] MARF Insert in b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b: '8cdbe925a0ce59c46516c931e1b84a9facf6626e6acc2c176d2d9850b01fd33c' = 'b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b0000000000000000' (...[])
DEBUG [1603139396.457] [src/chainstate/stacks/index/trie.rs:787] [ThreadId(9)] Next root hash is 60120ce35988c4b2052eb0a41fa7eda4e94dbc43fac83f7a447ad1b1f10d2fe8 (update_skiplist=false)
DEBUG [1603139396.457] [src/chainstate/stacks/index/marf.rs:853] [ThreadId(9)] MARF Insert in b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b: 'f4babc0ef7455166ed8559a53cabdcc1a67798d65a9d08b92d854869872212ba' = '03000000000000000000000000000000000000000000000000000000000000000000000000000000' (...[])
DEBUG [1603139396.459] [src/chainstate/stacks/index/trie.rs:851] [ThreadId(9)] Next root hash is 8d4e490c33099701aedde3ad7bdd52b1303526a84f80681e0f3e5379504ebeff (update_skiplist=false)
DEBUG [1603139396.459] [src/chainstate/stacks/index/marf.rs:853] [ThreadId(9)] MARF Insert in b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b: 'ee04b95e997d3560613e00a48d49273eb6763707b71be61d6d1e3a31c2cb740a' = 'e56fde2bbb789fab4ad51501fc7658fbcb440b1075815d86022c17e78f21c8830000000000000000' (...[])
DEBUG [1603139396.460] [src/chainstate/stacks/index/trie.rs:851] [ThreadId(9)] Next root hash is c03ebebc0b1b8d056756b34dcfd0abd119815ca6012a9f9485bce1cbc5f7cd4c (update_skiplist=false)
DEBUG [1603139396.461] [src/chainstate/stacks/index/marf.rs:853] [ThreadId(9)] MARF Insert in b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b: 'c3ba68392def1bdaccae0ad74d41e6dda7db795de41af363e041d3734c4f25ce' = '02000000000000000000000000000000000000000000000000000000000000000000000000000000' (...[])
DEBUG [1603139396.464] [src/chainstate/stacks/index/trie.rs:787] [ThreadId(9)] Next root hash is cc4eb11b44e7f0c3672894e34b65eca4a6e67d9c5110e0e1ebefd4fd766b37ab (update_skiplist=true)
DEBUG [1603139396.464] [src/chainstate/stacks/index/marf.rs:260] [ThreadId(9)] Opened e56fde2bbb789fab4ad51501fc7658fbcb440b1075815d86022c17e78f21c883 to b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b
DEBUG [1603139396.735] [src/chainstate/stacks/db/blocks.rs:3649] [ThreadId(9)] Process microblock bd8232849dd080426b97508013bf32e6fe446375f54498e302071344a5a7fd94
DEBUG [1603139396.735] [src/chainstate/stacks/db/transactions.rs:846] [ThreadId(9)] Process transaction cf110e240610394b815624530cd9f7e7ae4f6982c460ba6f063fa35fabbecf9e (SmartContract)
DEBUG [1603139398.148] [src/chainstate/stacks/index/marf.rs:853] [ThreadId(9)] MARF Insert in b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b: '79a61093847801451248f1cace6cbe3eebfa81feec74b63823271aa2ee4e689d' = 'e9775ed00c4c1a5bd67f2da686b417b1b8743065d296a3d503b21f27a6a45af50000000000000000' (...[])
DEBUG [1603139398.150] [src/chainstate/stacks/index/trie.rs:851] [ThreadId(9)] Next root hash is 443e0520e1e8dd0527f2f573e3b5c22c58a07e3d6a1b5bb46a158388fec1dcb0 (update_skiplist=false)
DEBUG [1603139398.151] [src/chainstate/stacks/index/marf.rs:853] [ThreadId(9)] MARF Insert in b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b: '797a79a1f66ee02a6e5540ca825223f7c685cf885f92121bee2032bf0343d8b2' = '3a6165fa600155c9ccc797b36e2b05cfcc944945cbf058291bdf0f3a13289f640000000000000000' (...[])
DEBUG [1603139398.152] [src/chainstate/stacks/index/trie.rs:851] [ThreadId(9)] Next root hash is 4cb66dcb5bf02242dfeaa1fc0ff1c58ee9267e97f5df128b47d05e2330abed1d (update_skiplist=false)
DEBUG [1603139398.153] [src/chainstate/stacks/index/marf.rs:853] [ThreadId(9)] MARF Insert in b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b: '7f046bed5597a3b8c0540e4f9be69cb455ffdd5c6f74cdd4646a239d8c8d5082' = '18d27566bd1ac66b2332d8c54ad43f7bb22079c906d05f491f3f07a28d5c69900000000000000000' (...[])
DEBUG [1603139398.156] [src/chainstate/stacks/index/trie.rs:787] [ThreadId(9)] Next root hash is 6e4437790c6d03ad026c488e216a591a497ba913cc31f920a570b6ffa5141804 (update_skiplist=true)
DEBUG [1603139398.160] [src/chainstate/stacks/db/transactions.rs:846] [ThreadId(9)] Process transaction 3e04ada5426332bfef446ba0a06d124aace4ade5c11840f541bf88e2e919faf6 (Coinbase)
DEBUG [1603139398.164] [src/chainstate/stacks/index/marf.rs:853] [ThreadId(9)] MARF Insert in b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b: '173c9bf7f7e2955c5515072450b136a35c56253db4538d6a87cad29c192095c5' = '3a6165fa600155c9ccc797b36e2b05cfcc944945cbf058291bdf0f3a13289f640000000000000000' (...[])
DEBUG [1603139398.166] [src/chainstate/stacks/index/trie.rs:851] [ThreadId(9)] Next root hash is 7ba56c8c5c42fb6252e82f522ce18811a566a7d4a05b5752033415d061489d00 (update_skiplist=false)
DEBUG [1603139398.166] [src/chainstate/stacks/index/marf.rs:853] [ThreadId(9)] MARF Insert in b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b: '50bb7bbbc5d8d6b5447010708db500655640b7b454c75c17394a348cf960c9c3' = '4e36e78cb8eb346375194c1de90113cab8680519945fec5bb7cd216eefb482be0000000000000000' (...[])
DEBUG [1603139398.169] [src/chainstate/stacks/index/trie.rs:851] [ThreadId(9)] Next root hash is d61af4e9d19ee2725af972eb736483da5bf74e0289822572f665b11eccd3c03b (update_skiplist=true)
DEBUG [1603139398.169] [src/chainstate/stacks/miner.rs:531] [ThreadId(9)] Include tx 3e04ada5426332bfef446ba0a06d124aace4ade5c11840f541bf88e2e919faf6
DEBUG [1603139398.169] [src/core/mempool.rs:514] [ThreadId(9)] Begin scanning transaction mempool at fcff47e4ad8f793a8fc0cb9f062968acf754acbe/1b82e31de5a46df0e464c2389624e42f77e9c52d5ca19dedac9616341bd128b2 height=2
DEBUG [1603139398.172] [src/core/mempool.rs:477] [ThreadId(9)] Will start scaning mempool at 57f5de67bebdd8dcc66aea02bc898dfe3c5bc7f5/4b02265bc6a22a325c252f572d5ec606dc10e67995b5afae7094d2ee6deb84f2 height=1 ts=1603139344
DEBUG [1603139399.220] [src/core/mempool.rs:552] [ThreadId(9)] Have 1 transactions at 57f5de67bebdd8dcc66aea02bc898dfe3c5bc7f5/4b02265bc6a22a325c252f572d5ec606dc10e67995b5afae7094d2ee6deb84f2 height=1 at or after 1603139344
DEBUG [1603139399.953] [src/chainstate/stacks/db/transactions.rs:846] [ThreadId(9)] Process transaction 27e29985d1f7dd4a5c1f3a8659ab896dfd322b45ee1514219e8f3b8439d878bd (SmartContract)
DEBUG [1603139401.185] [testnet/stacks-node/src/syncctl.rs:399] [ThreadId(4)] PoX watchdog: flat-attachable?: true, flat-processed?: true, estimated block-download time: 1s, estimated block-processing time: 1s
DEBUG [1603139401.185] [testnet/stacks-node/src/syncctl.rs:432] [ThreadId(4)] PoX watchdog: In steady-state; waiting until at least 1603139402 before burnchain synchronization
DEBUG [1603139401.192] [src/chainstate/stacks/index/marf.rs:853] [ThreadId(9)] MARF Insert in b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b: '0cc6ddd9bb5405ee34bb2ce823183921625f26fd3a1e42c9d5425d065b964e1f' = 'e9775ed00c4c1a5bd67f2da686b417b1b8743065d296a3d503b21f27a6a45af50000000000000000' (...[])
DEBUG [1603139401.195] [src/chainstate/stacks/index/trie.rs:787] [ThreadId(9)] Next root hash is 0f15af766ebf7ab22f310e88e35118fdcfa7ed874a8dd66dfe112864aa91de45 (update_skiplist=false)
DEBUG [1603139401.195] [src/chainstate/stacks/index/marf.rs:853] [ThreadId(9)] MARF Insert in b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b: 'c2d2c3207ef5a9f1ac1473e9342e478998bce632eaddc78686db4905639cc629' = '3a6165fa600155c9ccc797b36e2b05cfcc944945cbf058291bdf0f3a13289f640000000000000000' (...[])
DEBUG [1603139401.197] [src/chainstate/stacks/index/trie.rs:851] [ThreadId(9)] Next root hash is 31f845f62560052c26606b257b3852162bb91fc8a455c7b3c0b7a60683e4ee40 (update_skiplist=false)
DEBUG [1603139401.197] [src/chainstate/stacks/index/marf.rs:853] [ThreadId(9)] MARF Insert in b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b: 'd73722e6e4f00d7c6b47dc6527e27846ecf648d26c28276797fcc66d9c8b9ec6' = '18d27566bd1ac66b2332d8c54ad43f7bb22079c906d05f491f3f07a28d5c69900000000000000000' (...[])
DEBUG [1603139401.200] [src/chainstate/stacks/index/trie.rs:787] [ThreadId(9)] Next root hash is 2e74abf0c5af43ea7fccd0b4c3ae8d9d3a7fad50276149db6e076877f3abd992 (update_skiplist=true)
DEBUG [1603139401.306] [src/chainstate/stacks/miner.rs:531] [ThreadId(9)] Include tx 27e29985d1f7dd4a5c1f3a8659ab896dfd322b45ee1514219e8f3b8439d878bd
DEBUG [1603139402.690] [src/core/mempool.rs:552] [ThreadId(9)] Have 1 transactions at 57f5de67bebdd8dcc66aea02bc898dfe3c5bc7f5/4b02265bc6a22a325c252f572d5ec606dc10e67995b5afae7094d2ee6deb84f2 height=1 at or after 1603139347
DEBUG [1603139403.417] [src/chainstate/stacks/miner.rs:1019] [ThreadId(9)] Block budget exceeded on tx cf110e240610394b815624530cd9f7e7ae4f6982c460ba6f063fa35fabbecf9e
DEBUG [1603139404.787] [src/core/mempool.rs:552] [ThreadId(9)] Have 1 transactions at 57f5de67bebdd8dcc66aea02bc898dfe3c5bc7f5/4b02265bc6a22a325c252f572d5ec606dc10e67995b5afae7094d2ee6deb84f2 height=1 at or after 1603139351
DEBUG [1603139405.507] [src/chainstate/stacks/miner.rs:1019] [ThreadId(9)] Block budget exceeded on tx d3b4edd4dc5092ebc52e1f9451de8cfd98a83f8d9447dcc35bf31ee07f9cdedc
DEBUG [1603139406.960] [src/core/mempool.rs:552] [ThreadId(9)] Have 1 transactions at 57f5de67bebdd8dcc66aea02bc898dfe3c5bc7f5/4b02265bc6a22a325c252f572d5ec606dc10e67995b5afae7094d2ee6deb84f2 height=1 at or after 1603139355
DEBUG [1603139407.684] [src/chainstate/stacks/miner.rs:1019] [ThreadId(9)] Block budget exceeded on tx e5c4189c3296ee40f3a87171210468bd8b405a688996a7b438051f46f597c3da
DEBUG [1603139409.052] [src/core/mempool.rs:552] [ThreadId(9)] Have 1 transactions at 57f5de67bebdd8dcc66aea02bc898dfe3c5bc7f5/4b02265bc6a22a325c252f572d5ec606dc10e67995b5afae7094d2ee6deb84f2 height=1 at or after 1603139359
DEBUG [1603139409.782] [src/chainstate/stacks/miner.rs:1019] [ThreadId(9)] Block budget exceeded on tx a3059a569898e73765f162e280e47764ea38472ce4a69f2fed90af9d33251f95
DEBUG [1603139411.186] [src/core/mempool.rs:552] [ThreadId(9)] Have 1 transactions at 57f5de67bebdd8dcc66aea02bc898dfe3c5bc7f5/4b02265bc6a22a325c252f572d5ec606dc10e67995b5afae7094d2ee6deb84f2 height=1 at or after 1603139362
DEBUG [1603139411.907] [src/chainstate/stacks/miner.rs:1019] [ThreadId(9)] Block budget exceeded on tx 985731e1b813dadc38802715735c45e9be6642ce88460aa4d1037a85f26f7fde
DEBUG [1603139413.262] [src/core/mempool.rs:552] [ThreadId(9)] Have 1 transactions at 57f5de67bebdd8dcc66aea02bc898dfe3c5bc7f5/4b02265bc6a22a325c252f572d5ec606dc10e67995b5afae7094d2ee6deb84f2 height=1 at or after 1603139366
DEBUG [1603139414.023] [src/chainstate/stacks/miner.rs:1019] [ThreadId(9)] Block budget exceeded on tx 0fd4e3292a95b4bca3006b4adf847d73619e89596a8615c02014fffbc545b205
DEBUG [1603139415.497] [src/core/mempool.rs:552] [ThreadId(9)] Have 1 transactions at 57f5de67bebdd8dcc66aea02bc898dfe3c5bc7f5/4b02265bc6a22a325c252f572d5ec606dc10e67995b5afae7094d2ee6deb84f2 height=1 at or after 1603139370
DEBUG [1603139416.373] [src/chainstate/stacks/miner.rs:1019] [ThreadId(9)] Block budget exceeded on tx 1448e80f3376c44d4668a3699b46686a8d0e3ae99cf8cd118e3821a881c4a3bf
DEBUG [1603139417.750] [src/core/mempool.rs:552] [ThreadId(9)] Have 1 transactions at 57f5de67bebdd8dcc66aea02bc898dfe3c5bc7f5/4b02265bc6a22a325c252f572d5ec606dc10e67995b5afae7094d2ee6deb84f2 height=1 at or after 1603139374
DEBUG [1603139418.481] [src/chainstate/stacks/miner.rs:1019] [ThreadId(9)] Block budget exceeded on tx 2d9489c04a65407206591ad5c0010b5c84a17cd21d4cd23bf2e87fd5a59920c6
DEBUG [1603139419.851] [src/core/mempool.rs:552] [ThreadId(9)] Have 1 transactions at 57f5de67bebdd8dcc66aea02bc898dfe3c5bc7f5/4b02265bc6a22a325c252f572d5ec606dc10e67995b5afae7094d2ee6deb84f2 height=1 at or after 1603139378
DEBUG [1603139420.570] [src/chainstate/stacks/miner.rs:1019] [ThreadId(9)] Block budget exceeded on tx c5b15556b111e1c8afdf2d10a1813da0451e06a81fb9b0e0519e9c375423b35e
DEBUG [1603139420.961] [src/core/mempool.rs:409] [ThreadId(9)] Done scanning mempool -- no chain tips at height 0
INFO [1603139421.324] [src/chainstate/stacks/miner.rs:672] [ThreadId(9)] Miner: mined anchored block 1b627c92e734b2283ed902c5eb580d190e0efc135d697d54d736c31f6ca57c1d with 2 txs, parent block 1b82e31de5a46df0e464c2389624e42f77e9c52d5ca19dedac9616341bd128b2, state root = 2e74abf0c5af43ea7fccd0b4c3ae8d9d3a7fad50276149db6e076877f3abd992
DEBUG [1603139421.325] [src/vm/clarity.rs:493] [ThreadId(9)] Commit mined Clarity datastore to b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b
DEBUG [1603139421.325] [src/vm/database/marf.rs:337] [ThreadId(9)] commit_mined_block: (b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b->b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b)
DEBUG [1603139421.488] [src/chainstate/stacks/index/storage.rs:941] [ThreadId(9)] Flush: b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b to b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b.mined
DEBUG [1603139421.489] [src/chainstate/stacks/index/storage.rs:985] [ThreadId(9)] Flush: identifier of b40ffa12cd6cdec17a0cf0231bffb40d9271c319a5113e88792a3488765e6c6b.mined is 3
INFO [1603139421.614] [testnet/stacks-node/src/neon_node.rs:1017] [ThreadId(9)] Stacks block assembled: 1b627c92e734b2283ed902c5eb580d190e0efc135d697d54d736c31f6ca57c1d, with 2 txs, attempt 1
DEBUG [1603139421.618] [testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs:1115] [ThreadId(9)] BitcoinRPC builder: Some("neon-tester"):Some("neon-tester-pass")@http://127.0.0.1:8332/
DEBUG [1603139421.639] [testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs:747] [ThreadId(9)] Send transaction: "0100000001ff31724381c7bdffd3e4c0bbe3e5004b38c0431715e6ac9f23871f35a6409ae0000000008b483045022100b85834c218a8083d166c657379ae0594548898611cc6eafb52485bee7af9b21e0220421220dc6391d73260e7c97b833d2c09e5d538070f336d4ccc9902361615f498014104ee0b1602eb18fef7986887a7e8769a30c9df981d33c8380d255edef003abdcd243a0eb74afdf6740e6c423e62aec631519a24cf5b1d62bf8a3e06ddc695dcb77fdffffff030000000000000000536a4c5069645b1b627c92e734b2283ed902c5eb580d190e0efc135d697d54d736c31f6ca57c1d19764111c9ae6a6fea0f806076b4fd4e0439ca05099a91d99537955cf6d52225000000cd0001000000cb00010010270000000000001976a914000000000000000000000000000000000000000088ac74b5052a010000001976a9148bc5147525b8f477f0bc4522a88c8339b2494db588ac00000000"
INFO [1603139421.639] [testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs:617] [ThreadId(9)] Miner node: submitting leader_block_commit op - 04ee0b1602eb18fef7986887a7e8769a30c9df981d33c8380d255edef003abdcd243a0eb74afdf6740e6c423e62aec631519a24cf5b1d62bf8a3e06ddc695dcb77
DEBUG [1603139421.640] [testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs:1115] [ThreadId(9)] BitcoinRPC builder: Some("neon-tester"):Some("neon-tester-pass")@http://127.0.0.1:8332/
DEBUG [1603139421.642] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure
DEBUG [1603139421.645] [testnet/stacks-node/src/neon_node.rs:906] [ThreadId(9)] Mining tenure's last consensus hash: fcff47e4ad8f793a8fc0cb9f062968acf754acbe (height 205 hash 4f4ba280b1e24625f8364722a172d7359bc122db77edaa0bb6e63f582101ae0d), stacks tip consensus hash: fcff47e4ad8f793a8fc0cb9f062968acf754acbe (height 205 hash 4f4ba280b1e24625f8364722a172d7359bc122db77edaa0bb6e63f582101ae0d)
DEBUG [1603139421.647] [testnet/stacks-node/src/neon_node.rs:956] [ThreadId(9)] Stacks tip is unchanged since we last tried to mine a block (fcff47e4ad8f793a8fc0cb9f062968acf754acbe/1b627c92e734b2283ed902c5eb580d190e0efc135d697d54d736c31f6ca57c1d at height 3 with 2 txs, in 4f4ba280b1e24625f8364722a172d7359bc122db77edaa0bb6e63f582101ae0d at burn height 205)
```

As you can see, this test is simply generating transactions that take a long time to evaluate, thereby causing the first `RunTenure` to take a while to process.  But it finishes before the next Bitcoin block is produced!  Once it is processed, the subsequent `RunTenure`s that had been generated for the same Bitcoin block height process quickly, because the Stacks chain tip hasn't changed (i.e. no new blocks have been discovered from any neighbors, this block hasn't won sortition yet, and the burnchain has not advanced).

Just to make sure I'm not crazy, I re-ran this test to make `RunTenure`'s debug output print the bitcoin block height and burn header hash it's working on.  I also updated `relayer_run_tenure()` to only proceed if the given `burn_block` is still currently equal to the canonical burnchain tip (i.e. no subsequent sortitions were processed by the time this `RunTenure` gets processed).  If the `RunTenure` was late -- i.e. if the burnchain advanced between when it was generated and when it was processed -- we'd see the node print out `New canonical burn chain tip detected`.  The resulting test run confirms my explanation above -- the node is processing multiple `RelayerDirective::RunTenure`s per Bitcoin block as expected, and the first one for a given Bitcoin block is the one that generates the Stacks block.  Subsequent ones do nothing, since the canonical Stacks chain tip has not changed.  Output:

```
$ egrep 'Proceeding to mine block|Run tenure|New canonical burn chain tip' /tmp/out.txt
DEBUG [1603160274.274] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 202. Proceeding to mine blocks
DEBUG [1603160276.378] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 203. Proceeding to mine blocks
DEBUG [1603160276.879] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 203 (34b62ebf26fe3cd1432017d50f07f37015f5475cc113cb1735f20e1e265cf992)
DEBUG [1603160280.826] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160282.136] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160283.026] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160283.526] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160285.310] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160285.811] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160287.460] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160287.960] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160289.684] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160290.185] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160291.820] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160292.320] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160293.970] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160294.758] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160296.100] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160296.601] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160298.191] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160298.692] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160300.228] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160300.729] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160302.330] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160302.830] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160304.592] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160305.092] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160306.741] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160307.241] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160308.849] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160309.589] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160310.999] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160311.500] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160313.196] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160313.698] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160315.423] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160315.923] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160317.464] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160317.964] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160319.566] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 204. Proceeding to mine blocks
DEBUG [1603160320.066] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 204 (2d7642fe8508b01463e45e923c9dd8da879fecb1ddfc8c9966f594a88065d10b)
DEBUG [1603160322.689] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603160324.645] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603160326.862] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603160328.921] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603160330.985] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603160333.131] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603160334.542] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 205 (30d5f274e5419bc31493aae73d3e709365e3f34f98294677b972a8b4207dcb41)
DEBUG [1603160335.176] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603160337.318] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603160339.411] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603160341.606] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603160343.854] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603160346.000] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603160348.195] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603160350.267] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603160352.462] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603160354.585] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603160357.038] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603160359.160] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 205. Proceeding to mine blocks
DEBUG [1603160360.424] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 205 (30d5f274e5419bc31493aae73d3e709365e3f34f98294677b972a8b4207dcb41)
DEBUG [1603160360.429] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 205 (30d5f274e5419bc31493aae73d3e709365e3f34f98294677b972a8b4207dcb41)
DEBUG [1603160360.434] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 205 (30d5f274e5419bc31493aae73d3e709365e3f34f98294677b972a8b4207dcb41)
DEBUG [1603160360.439] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 205 (30d5f274e5419bc31493aae73d3e709365e3f34f98294677b972a8b4207dcb41)
DEBUG [1603160360.445] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 205 (30d5f274e5419bc31493aae73d3e709365e3f34f98294677b972a8b4207dcb41)
DEBUG [1603160360.450] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 205 (30d5f274e5419bc31493aae73d3e709365e3f34f98294677b972a8b4207dcb41)
DEBUG [1603160360.455] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 205 (30d5f274e5419bc31493aae73d3e709365e3f34f98294677b972a8b4207dcb41)
DEBUG [1603160360.461] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 205 (30d5f274e5419bc31493aae73d3e709365e3f34f98294677b972a8b4207dcb41)
DEBUG [1603160360.468] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 205 (30d5f274e5419bc31493aae73d3e709365e3f34f98294677b972a8b4207dcb41)
DEBUG [1603160360.473] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 205 (30d5f274e5419bc31493aae73d3e709365e3f34f98294677b972a8b4207dcb41)
DEBUG [1603160360.478] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 205 (30d5f274e5419bc31493aae73d3e709365e3f34f98294677b972a8b4207dcb41)
DEBUG [1603160360.483] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 205 (30d5f274e5419bc31493aae73d3e709365e3f34f98294677b972a8b4207dcb41)
DEBUG [1603160360.489] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 205 (30d5f274e5419bc31493aae73d3e709365e3f34f98294677b972a8b4207dcb41)
DEBUG [1603160360.495] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 205 (30d5f274e5419bc31493aae73d3e709365e3f34f98294677b972a8b4207dcb41)
DEBUG [1603160360.500] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 205 (30d5f274e5419bc31493aae73d3e709365e3f34f98294677b972a8b4207dcb41)
DEBUG [1603160360.505] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 205 (30d5f274e5419bc31493aae73d3e709365e3f34f98294677b972a8b4207dcb41)
DEBUG [1603160360.510] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 205 (30d5f274e5419bc31493aae73d3e709365e3f34f98294677b972a8b4207dcb41)
DEBUG [1603160361.723] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160364.031] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160366.098] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160368.256] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160370.311] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160372.563] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160374.762] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160376.027] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160376.812] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160378.909] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160380.898] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160383.000] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160385.037] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160387.081] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160389.379] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160391.431] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160393.528] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160395.500] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160397.619] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160399.946] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 206. Proceeding to mine blocks
DEBUG [1603160400.007] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160400.013] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160400.018] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160400.023] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160400.029] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160400.034] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160400.039] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160400.045] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160400.050] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160400.055] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160400.060] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160400.067] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160400.072] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160400.077] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160400.082] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160400.087] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160400.093] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160400.446] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 206 (7d54be50d49ac5a977d39cfc50c03d393dd4bc47a68d038ac6e98bea3196a97a)
DEBUG [1603160402.680] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603160404.881] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603160406.853] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603160408.981] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603160411.067] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603160411.226] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 207 (390b03b975526d1ecdfe9750b041a18e60c4f470feb0c2ac2d3dce9dfb7892fd)
DEBUG [1603160413.239] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603160415.318] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603160417.462] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603160419.482] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603160421.547] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603160423.776] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603160425.834] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603160427.882] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603160429.875] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603160431.853] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603160433.891] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
DEBUG [1603160436.050] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 207 (390b03b975526d1ecdfe9750b041a18e60c4f470feb0c2ac2d3dce9dfb7892fd)
DEBUG [1603160436.056] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 207 (390b03b975526d1ecdfe9750b041a18e60c4f470feb0c2ac2d3dce9dfb7892fd)
DEBUG [1603160436.062] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 207 (390b03b975526d1ecdfe9750b041a18e60c4f470feb0c2ac2d3dce9dfb7892fd)
DEBUG [1603160436.067] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 207 (390b03b975526d1ecdfe9750b041a18e60c4f470feb0c2ac2d3dce9dfb7892fd)
DEBUG [1603160436.073] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 207 (390b03b975526d1ecdfe9750b041a18e60c4f470feb0c2ac2d3dce9dfb7892fd)
DEBUG [1603160436.078] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 207 (390b03b975526d1ecdfe9750b041a18e60c4f470feb0c2ac2d3dce9dfb7892fd)
DEBUG [1603160436.082] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 207 (390b03b975526d1ecdfe9750b041a18e60c4f470feb0c2ac2d3dce9dfb7892fd)
DEBUG [1603160436.088] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 207 (390b03b975526d1ecdfe9750b041a18e60c4f470feb0c2ac2d3dce9dfb7892fd)
DEBUG [1603160436.093] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 207 (390b03b975526d1ecdfe9750b041a18e60c4f470feb0c2ac2d3dce9dfb7892fd)
DEBUG [1603160436.098] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 207 (390b03b975526d1ecdfe9750b041a18e60c4f470feb0c2ac2d3dce9dfb7892fd)
DEBUG [1603160436.103] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 207 (390b03b975526d1ecdfe9750b041a18e60c4f470feb0c2ac2d3dce9dfb7892fd)
DEBUG [1603160436.109] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 207 (390b03b975526d1ecdfe9750b041a18e60c4f470feb0c2ac2d3dce9dfb7892fd)
DEBUG [1603160436.113] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 207 (390b03b975526d1ecdfe9750b041a18e60c4f470feb0c2ac2d3dce9dfb7892fd)
DEBUG [1603160436.118] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 207 (390b03b975526d1ecdfe9750b041a18e60c4f470feb0c2ac2d3dce9dfb7892fd)
DEBUG [1603160436.124] [testnet/stacks-node/src/neon_node.rs:563] [ThreadId(9)] Relayer: Run tenure at height 207 (390b03b975526d1ecdfe9750b041a18e60c4f470feb0c2ac2d3dce9dfb7892fd)
DEBUG [1603160436.197] [testnet/stacks-node/src/run_loop/neon.rs:279] [ThreadId(4)] Synchronized full burnchain up to height 207. Proceeding to mine blocks
```

So, unless you can provide evidence to the contrary, I think it's safe to conclude that the miner is not falling behind the burnchain.  In fact, I'll push these recent updates to make _sure_ that the miner doesn't run if the `RunTenure`'s given `BlockSnapshot` is no longer the canonical burnchain snapshot.

> but the logs you showed do not demonstrate the race condition described

That's because there is currently no easy way to test this with just a single miner -- you need at least two miners.  I'll go ahead and set up a separate public testnet where miners do not wait at all before trying to mine a block, and you can spin up a miner of your own and observe in its logs that it is (1) encountering this race condition and (2) correctly handling it by re-mining its block off of the new chain tip and RBF'ing its block-commit.

Reply:
> This is no longer true in this PR -- this PR intentionally changed this behavior so that a single sortition may generate multiple RunTenures (i.e. at a rate of once per poll_time_secs, as long as poll_time_secs isn't unreasonably small).

Ah, okay, I see this now. I think my confusion stemmed from the fact that two behaviors changed (in the PR that added syncctl, the bitcoin controller's sync behavior changed: it used to block until the burn block advanced) and then the neon run loop changed in this PR. I can see that this is now running multiple times per sortition.